drivers/gnss/Kconfig.u_blox_m10

```diff
@@ -0,0 +1,30 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config GNSS_U_BLOX_M10
+	bool "U-BLOX M10 GNSS Module"
+	default y
+	depends on GNSS
+	depends on DT_HAS_U_BLOX_M10_ENABLED
+	select UART_INTERRUPT_DRIVEN
```

Comment:
The UART API type is specified by the application, the driver should only select `MODEM_BACKEND_UART`